librdkafka/Sanity/consumer-producer/runtest.sh

```diff
@@ -0,0 +1,120 @@
+#!/bin/bash
+# vim: dict+=/usr/share/beakerlib/dictionary.vim cpt=.,w,b,u,t,i,k
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+#   runtest.sh of /CoreOS/librdkafka/Sanity/consumer-producer
+#   Description: sanity test for rdkafka library
+#   Author: Attila Lakatos <alakatos@redhat.com>
+#
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+#   Copyright (c) 2023 Red Hat, Inc.
+#
+#   This copyrighted material is made available to anyone wishing
+#   to use, modify, copy, or redistribute it subject to the terms
+#   and conditions of the GNU General Public License version 2.
+#
+#   This program is distributed in the hope that it will be
+#   useful, but WITHOUT ANY WARRANTY; without even the implied
+#   warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+#   PURPOSE. See the GNU General Public License for more details.
+#
+#   You should have received a copy of the GNU General Public
+#   License along with this program; if not, write to the Free
+#   Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+#   Boston, MA 02110-1301, USA.
+#
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+# Include Beaker environment
+# . /usr/bin/rhts-environment.sh || :
+. /usr/share/beakerlib/beakerlib.sh || exit 1
+
+PACKAGE="rsyslog"
+CONTROLLERID=123
+
+# This test focuses on creating a producer-consumer environment and estabilishing
+# exactly one consumer(server) and one producer(client). First, the producer will
+# start generating messages, which will be then processed/received by the consumer.
+# The test checks if all messages have arrived successfully and if the rd_kafka_controllerid
+# symbol is available in the dynamically linked shared object library and if it works as expected.
+
+rlJournalStart && {
+  rlPhaseStartSetup && {
+    rlRun "rlImport --all" 0 "Import libraries" || rlDie "cannot continue"
+    tcfRun "rlCheckMakefileRequires"
+    CleanupRegister 'rlRun "rlSEPortRestore"'
+
+    rlRun "rlSEPortAdd tcp 9092 syslogd_port_t"
+    rlRun "TmpDir=\$(mktemp -d)" 0 "Creating tmp directory"
+    CleanupRegister "rlRun 'rm -r $TmpDir' 0 'Removing tmp directory'"
+    CleanupRegister 'rlRun "popd"'
+    rlRun "ls -al"
+    rlRun "cp Makefile kafka-prog.c $TmpDir"
+    rlRun "pushd $TmpDir"
+    CleanupRegister 'rlRun "rlFileRestore"'
+    rlRun "rlFileBackup --clean /tmp/kafka-logs /tmp/zookeeper"
+    rlRun "rm -rf /tmp/kafka-logs /tmp/zookeeper"
+
+    rlRun "rlDownload kafka_2.11-2.1.0.tgz http://download.eng.bos.redhat.com/qa/rhts/lookaside/kafka_2.11-2.1.0.tgz"
```

Comment:
we need to use different location for this in a public space
I already (mis)used the wiki in such a way https://github.com/RedHat-SP-Security/tests/wiki/sideload/elasticsearch-1.3.1.noarch.rpm.
But recently I noticed somewhere there might be an official lookaside for these cases.

Reply:
I've tried it and the wiki does not allow to upload tarballs, so I modified the hyperlink to point to the official apache kafka release page.

Reply:
The wiki page has its own git repo so it needs to be committed through it. Anyway, the direct link might be better, after all.